plugins/content/local/locks.go

```diff
@@ -30,17 +30,23 @@ type lock struct {
 	since time.Time
 }
 
+type key struct {
+	// Scope ref locking to root directories.
+	// E.g. In buildkit, each worker has its own content store.
+	root, ref string
+}
+
 var (
 	// locks lets us lock in process
-	locks   = make(map[string]*lock)
+	locks   = make(map[key]*lock)
```

Comment:
Why isn't this map just a property of `store{}` instead of global?

Reply:
I don't see any reason why, there should not be multiple instances of a store for the same root. +1 for not using global variables when not needed.

Reply:
SG, updating now - will ping when done.

Reply:
PTAL